### PR TITLE
fix(eks): node-termination-handler cannot be scheduled in default namespace

### LINKED ISSUE #6990: node-termination-handler installed into 'default' namespace, not kube-system
https://github.com/aws/aws-cdk/blob/f4a41d1c68a42e796294692ef00c1fbc83d5dcff/packages/%40aws-cdk/aws-eks/lib/spot-interrupt-handler.ts#L15
(and similar lines in that file)

Despite the name, a less surprising default namespace would be `kube-system`.  The actual `default` namespace is a legacy and almost never used in Kubernetes.

### PR DESCRIPTION
**Pull request by pahud**
### Commit Message

`node-termination-handler` now deployed with Helm Chart([current version 1.7.3](https://github.com/aws/eks-charts/blob/e421230a2ad39c53e287ef6222d4438932460ebd/stable/aws-node-termination-handler/Chart.yaml#L4-L5)) in the `kube-system` namespace

fixeds #6990 

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
copy @eladb

File: packages/@aws-cdk/aws-eks/lib/spot-interrupt-handler.ts [hunk 1]
```diff
@@ -3,173 +3,13 @@ export enum LifecycleLabel {
...
-      metadata: {
-        name: "node-termination-handler",
-        namespace: "default"
-      },
-      subjects: [
-        {
-          kind: "ServiceAccount",
-          name: "node-termination-handler",
-          namespace: "default"
+export function spotInterruptHandler() {
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/spot-interrupt-handler.ts:6]: I think this file is now redundant. You can just move these attributes to the main file.

File: packages/@aws-cdk/aws-eks/lib/spot-interrupt-handler.ts [hunk 2]
```diff
@@ -3,173 +3,13 @@ export enum LifecycleLabel {
...
+  return {
+        chart: 'aws-node-termination-handler',
+        repository: 'https://aws.github.io/eks-charts',
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/spot-interrupt-handler.ts:9]: Does it make sense to specify a version here so this doesn't break unexpectedly?

Comment (pahud) [reply packages/@aws-cdk/aws-eks/lib/spot-interrupt-handler.ts:9]: You are right. Will pin the current chart version to [1.7.3](https://github.com/aws/eks-charts/blob/e421230a2ad39c53e287ef6222d4438932460ebd/stable/aws-node-termination-handler/Chart.yaml#L4-L5).

File: packages/@aws-cdk/aws-eks/package.json [hunk 3]
```diff
@@ -108,8 +108,8 @@
       "resource-attribute:@aws-cdk/aws-eks.Cluster.clusterSecurityGroupId",
       "resource-attribute:@aws-cdk/aws-eks.Cluster.clusterSecurityGroupId",
       "docs-public-apis:@aws-cdk/aws-eks.NodegroupAmiType.AL2_X86_64",
-      "docs-public-apis:@aws-cdk/aws-eks.NodegroupAmiType.AL2_X86_64_GPU"
-
+      "docs-public-apis:@aws-cdk/aws-eks.NodegroupAmiType.AL2_X86_64_GPU",
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/package.json:111]: Why do we need these two exclusions? Let’s just fix them

Comment (pahud) [reply packages/@aws-cdk/aws-eks/package.json:111]: Hi @eladb 

I got this error when I run `yarn awslint`

>error: [awslint:no-unused-type:@aws-cdk/aws-eks.LifecycleLabel] type or enum is not used by exported classes (declared at lib/cluster.ts:1001)

However, I was actually using this enum in the `Cluster` class.  
https://github.com/aws/aws-cdk/blob/9b515ab4f7b135f7431ef81b25ba690d317e69ee/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L597

Can you advise how to fix this warning?

Comment (eladb) [reply packages/@aws-cdk/aws-eks/package.json:111]: It's used but not in a public API. Just remove the `export` keyword from the enum definition:

~`export`~`enum LifecycleLabel {`

**Comment by eladb** [comment]
@pahud can you please resolve the conflicts?

**Comment by pahud** [comment]
Sure. Will get back to this after 1 hour.

**Pull request merged**